From 9d730457793473d4f137aa8e6322d165d0ae6368 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <zhengxingda@iscas.ac.cn>
Date: Sun, 19 Apr 2026 02:12:40 +0800
Subject: [PATCH 311/334] AOSCOS: arm64: dts: mediatek: mt8188-geralt: enable
 Wi-Fi card

The mainline pcie-mediatek-gen3 driver does not have code managing
downstream device power / reset.

As the Wi-Fi card on ciri is a fixed device, set the related regulator
to always-on and use GPIO hog to set the status of its reset pin.

Signed-off-by: Icenowy Zheng <zhengxingda@iscas.ac.cn>
Signed-off-by: Icenowy Zheng <uwu@icenowy.me>
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 arch/arm64/boot/dts/mediatek/mt8188-geralt.dtsi | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/arch/arm64/boot/dts/mediatek/mt8188-geralt.dtsi b/arch/arm64/boot/dts/mediatek/mt8188-geralt.dtsi
index 4cb23595d17b4..80abbe3ca75a7 100644
--- a/arch/arm64/boot/dts/mediatek/mt8188-geralt.dtsi
+++ b/arch/arm64/boot/dts/mediatek/mt8188-geralt.dtsi
@@ -568,6 +568,11 @@ &mt6359codec {
 	mediatek,mic-type-2 = <2>; /* DMIC */
 };
 
+&mt6359_vcn18_ldo_reg {
+	/* Used by WLAN */
+	regulator-always-on;
+};
+
 &mt6359_vcore_buck_reg {
 	regulator-always-on;
 };
@@ -1169,6 +1174,12 @@ pins-en-pp3300-wlan {
 			output-low;
 		};
 	};
+
+	wlan-reset-hog {
+		gpio-hog;
+		gpios = <145 GPIO_ACTIVE_HIGH>;
+		output-high;
+	};
 };
 
 &pmic {
-- 
2.55.0

